infra/src/main/kotlin/fi/fta/geoviite/infra/publication/Publication.kt

```diff
@@ -45,6 +45,7 @@ data class PublicationChange(
     val oldValue: String?,
     val newValue: String?,
     val remark: PublicationChangeRemark?,
+    val enumKey: String? = null,
```

Comment:
Frontilla ei tiedetä mistä propsusta on kyse, mutta enumit pitää silti kääntää -> palautetaan enumin identifioiva käännösavain bäkkäriltä

Reply:
Enum-arvot täytyy lokalisoida jotta ne saadaan näytettyä frontilla järkevästi. Meillä on frontilla käännökset enumeille (löytyvät käännösavaimen `enum.<enumin nimi>.<enumin arvo>` takaa.) Tämä kuvastaa käytännössä tuota keskiosuutta. Rakenteellisesti olisi ehkä ollut oikeampaa yhdistää tämä suoraan arvokenttiin, mutta silloin se monistuisi, kun uusi ja vanha arvo kuljetetaan kuitenkin erillään. Myönnettäköön että tällä on nyt huono nimi ja tyypitys, tämä olisi ehkä auennut paremmin jos nimenä olisi ollut esim. `enumLocalizationKey`, tai edes tyyppinä `LocalizationKey`.

Mietin tätä koko `PublicationChange`-luokan rakennetta ylipäätään vielä, ja voi olla että se elää johonkin suuntaan (tyypitys tulee elämään varmasti.) Tää on ollut vielä vähän tarkoituksellakin sinnepäin, kun tämä siirtorakenne ei oo ollut kriittisimmästä päästä ongelmia ja tälläkin saadaan jutut toimimaan.

Reply:
Hmm, aivan. Eli propKey:llä haetaan label ja enumKey:llä (jos se on) lokalisoidaan itse arvot. Makes sense.